whole-tale/gwvolman
# Significantly restrict privileges required to run gwvolman

This PR allows us to finally stop running everything as root by default. Noteworthy changes:

1. Mounting the entire host's `/` is no longer necessary. We just need a regular folder that serves as a root for mounts and is mount-binded with `rshared` option. Its location inside the container is relative to `VOLUMES_ROOT`.
2. Mounting `/usr/bin/docker` is no longer necessary. Stats collection is taken care via regular api access to docker's socket. See `gwvolman/lib/stats_collector.py` for details.
3. Dropped redundant `Pipfile` setup that was only used by tests, decreasing the number of places that had to be updated during releases. 
4. Updated cicircle setup to use venv and as result everything is 5 times faster...

## How to test?
1. Ensure you test it in combination with https://github.com/whole-tale/deploy-dev/pull/57
1. Run a Tale with external data attached. Confirm filesystems are visible and operational.
2. Run a Recorded Run. Confirm that `.docker_stats` contains the same info as before.
